src/winmain.cpp

```diff
@@ -917,6 +919,42 @@ bool runInstaller(const wstring& app2runPath, const wstring& binWindowsClassName
 	return true;
 }
 
+// Returns a folder suitable for exe installer download destination.
+// In case of error, shows a message box and returns an empty string.
+std::wstring getDestDir(const GupNativeLang& nativeLang, const GupParameters& gupParams)
+{
+	const wchar_t* envVar = _wgetenv(L"TEMP");
+	if (envVar)
+		return envVar;
+	envVar = _wgetenv(L"TMP");
+	if (envVar)
+		return envVar;
+
+	envVar = _wgetenv(L"AppData");
+	if (envVar)
+	{
+		std::wstring result = envVar;
+		result += L"\\Notepad++";
+		if (::PathFileExists(result.c_str()))
+			return result;
+	}
+
+	LPWSTR path = nullptr;
+	const HRESULT hr = ::SHGetKnownFolderPath(FOLDERID_Downloads, 0, nullptr, &path);
+	if (SUCCEEDED(hr))
+	{
+		std::wstring result = path;
+		CoTaskMemFree(path);
+		return result;
+	}

```

Comment:
Use Win32 API instead of COM, or remove above part completely.

Reply:
This is the only correct way to free the memory returned by `SHGetKnownFolderPath`.
I'll add `ole32.lib` to linker inputs. This is how it's done in Notepad++ ARM64 configuration.

Reply:
Removed.